gdnative-core/src/core_types/color.rs

```diff
@@ -48,6 +65,157 @@ impl Color {
         }
     }
 
+    #[inline]
+    pub fn blend(&self, other: &Color) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_blend)(self.sys(), other.sys()) })
+    }
+
+    #[inline]
+    pub fn contrasted(&self) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_contrasted)(self.sys()) })
+    }
+
+    #[inline]
+    pub fn darkened(&self, amount: f32) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_darkened)(self.sys(), amount) })
+    }
+
+    #[inline]
+    pub fn gray(&self) -> f32 {
+        // Implemented as described in godot docs
+        (self.r + self.b + self.g) / 3.0
+    }
+
+    #[inline]
+    pub fn inverted(&self) -> Color {
+        // Implementation as described in godot docs.
+        Color {
+            r: 1.0f32 - self.r,
+            g: 1.0f32 - self.g,
+            b: 1.0f32 - self.b,
+            a: self.a,
+        }
+    }
+
+    #[inline]
+    pub fn to_html(&self, with_alpha: bool) -> GodotString {
+        GodotString::from_sys(unsafe { (get_api().godot_color_to_html)(self.sys(), with_alpha) })
+    }
+
+    /// Returns the reverse of the RGBA32 byte representation for this color where each byte represents a component of the ABGR profile.
+    /// This is the byte information used when storing this color as a part of a texture.
+    /// # Endianness
+    /// On big endian architecture this is stored in ABGR byte order
+    /// On little endian machines this is stored in RGBA byte order
+    /// # Example
+    /// `0x00FF7FFF` would be the equivalent to `Color::rgba(1.0, 0.5, 1.0, 0.0)`
+    #[inline]
+    pub fn to_abgr32(&self) -> u32 {
+        let mut c: u32 = (self.a * 255_f32) as u32;
+        c <<= 8;
+        c |= (self.b * 255_f32) as u32;
+        c <<= 8;
+        c |= (self.g * 255_f32) as u32;
+        c <<= 8;
+        c |= (self.r * 255_f32) as u32;
+        c
```

Comment:
`255_f32` can simply be `255.0` with type inference, no? Or does clippy complain here?

There's no need for mutability, the expression can be simplified to the following (if needed, parentheses):
```rust
  (self.a * 255.0) as u32 << 24;
| (self.b * 255.0) as u32 << 16;
| (self.g * 255.0) as u32 << 8;
| (self.r * 255.0) as u32;
```
That's also clearer regarding offset of each component.

Reply:
Sounds like some good changes to me. I'll correct those.